### PR TITLE
LSR-851 Image Gallery Screenlet Android

### PR DESCRIPTION
**Pull request by victorg1991**

### CODE REVIEW

File: android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/gallery/grid/GridGalleryView.java [hunk 1]
```diff
@@ -103,6 +103,13 @@ protected int getItemProgressLayoutId() {
 	}
 
 	@Override
+	public void showStartOperation(String actionName) {
+		if(actionName == null){
```
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/gallery/grid/GridGalleryView.java]: why?

File: android/library/src/main/java/com/liferay/mobile/screens/gallery/interactor/upload/GalleryUploadService.java [hunk 2]
```diff
@@ -29,17 +32,39 @@
  */
 public class GalleryUploadService extends IntentService {
 
+	private boolean shouldCancel = false;
```
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/gallery/interactor/upload/GalleryUploadService.java]: default values are not usually (in common code conventions) specified explicitely

File: android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/gallery/UploadProgressView.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+	public void hide() {
+		uploadCount = 0;
+		images.clear();
+
+		setVisibility(GONE);
+	}
+
+	private void updateText() {
+		if (uploadCount == 1) {
```
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/gallery/UploadProgressView.java]: formatting

File: android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/gallery/grid/GridGalleryView.java [hunk 4]
```diff
@@ -76,6 +82,45 @@ public void addImage(ImageEntry imageEntry) {
...
+
+		uploadProgressView.setVisibility(VISIBLE);
+		uploadProgressView.addUpload(picturePath);
+	}
+
+	private void createProgressView() {
+		uploadProgressView =
+			(UploadProgressView) LayoutInflater.from(getContext()).inflate(R.layout.gallery_progress_view_default, null);
+
+		int height = getPixelsFromDp(60);
```
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/gallery/grid/GridGalleryView.java]: why?

File: android/library/src/main/java/com/liferay/mobile/screens/gallery/GalleryScreenlet.java [hunk 5]
```diff
@@ -184,6 +186,7 @@ public void onPicturePathReceived(String picturePath) {
 		if (getListener() != null) {
 			getListener().onImageUploadStarted();
 		}
+		getViewModel().imageUploadStart(picturePath);
```
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/gallery/GalleryScreenlet.java]: before listener

File: android/library/src/main/java/com/liferay/mobile/screens/gallery/BaseDetailUploadActivity.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * @author Víctor Galán Grande
+ */
+public class BaseDetailUploadActivity extends AppCompatActivity {
+
+	public static final String SCREENLET_ID_KEY = "SCREENLET_ID_KEY";
+	public static final String PICTURE_PATH_KEY = "PICTURE_PATH_KEY";
+
+	public void finishActivityAndstartUpload(int screenletId, String picturePath, String title, String description) {
```
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/gallery/BaseDetailUploadActivity.java]: Uhm, we are just providing this... let's talk tomorrow

File: android/library/src/main/res/layout/gallery_default_slideshow.xml [hunk 7]
```diff
@@ -3,7 +3,8 @@
     xmlns:android="http://schemas.android.com/apk/res/android"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    android:orientation="vertical">
+    android:orientation="vertical"
+    android:background="#000">
```
Comment (nhpatt) [inline android/library/src/main/res/layout/gallery_default_slideshow.xml]: android color black

File: android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/gallery/slideshow/SlideshowGalleryAdapter.java [hunk 8]
```diff
@@ -42,7 +42,7 @@ public SlideshowGalleryViewHolder(View view, BaseListAdapterListener listener) {
 		}
 
 		public void bind(ImageEntry entry) {
-			PicassoScreens.load(entry.getImageUrl()).placeholder(R.drawable.default_placeholder_image).into(_imageView);
+			PicassoScreens.load(entry.getImageUrl()).placeholder(R.drawable.default_placeholder_image).resize(900, 900).centerInside().into(_imageView);
```
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/gallery/slideshow/SlideshowGalleryAdapter.java]: 900? to constant

File: android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/gallery/slideshow/SlideshowLayout.java [hunk 9]
```diff
@@ -56,43 +55,36 @@ public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
...
-				v2.setTranslationY(200);
-				v2.setAlpha(0.5f);
+				View view = recyclerView.getChildAt(2);
+				assignDefaultValues(view);
 			}
-
 		}
 	}
 
+	private void assignDefaultValues(View view) {
```
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/gallery/slideshow/SlideshowLayout.java]: 👏

**Comment by nhpatt** [comment]
Apart from those things LGTM :)

File: android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/gallery/grid/GridGalleryView.java (hunk 1 above)
Comment (victorg1991) [inline android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/gallery/grid/GridGalleryView.java]: Fixed in 205d467

File: android/library/src/main/java/com/liferay/mobile/screens/gallery/interactor/upload/GalleryUploadService.java (hunk 2 above)
Comment (victorg1991) [inline android/library/src/main/java/com/liferay/mobile/screens/gallery/interactor/upload/GalleryUploadService.java]: Fixed in 	35e9274

File: android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/gallery/grid/GridGalleryView.java (hunk 4 above)
Comment (victorg1991) [inline android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/gallery/grid/GridGalleryView.java]: Fixed in 	9a9e0ea

File: android/library/src/main/java/com/liferay/mobile/screens/gallery/GalleryScreenlet.java (hunk 5 above)
Comment (victorg1991) [inline android/library/src/main/java/com/liferay/mobile/screens/gallery/GalleryScreenlet.java]: Fixed in 068d3f6

File: android/library/src/main/res/layout/gallery_default_slideshow.xml (hunk 7 above)
Comment (victorg1991) [inline android/library/src/main/res/layout/gallery_default_slideshow.xml]: fixed in 	4371779

File: android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/gallery/slideshow/SlideshowGalleryAdapter.java (hunk 8 above)
Comment (victorg1991) [inline android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/gallery/slideshow/SlideshowGalleryAdapter.java]: fixed in	c316959

**Pull request merged**